pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -525,6 +579,7 @@ func (w *wlReconciler) setupWithManager(mgr ctrl.Manager) error {
 		Named("multikueue_workload").
 		For(&kueue.Workload{}).
 		WatchesRawSource(source.Channel(w.clusters.wlUpdateCh, syncHndl)).
+		Watches(&kueue.MultiKueueConfig{}, &configHandler{client: w.client}).
```

Comment:
Maybe a bit out of scope, but worth asking:

`MultiKueueConfig` (config) and `MultiKueueCluster` (cluster) are separate. It’s possible for `config` to reference a new cluster **before** the cluster object is created/updated. Since we’re not watching for `cluster` changes, this could still require workload/job re-creation.

A simple repro: remove the `cluster` object (without touching config), submit a job, then add the cluster object back.

Should we be concerned about this in the context of this PR?